File: Solution/【003】无重复字符的最长子串/无重复字符的最长子串.py
'''
（1） len(s) 表示字符串的长度
（2） for i in range(len(s)) 用于迭代，要记住这个
（3） s[i] in dict    相当于直接用数组索引 表示 java中的 charAt()  并且 in dict 是只判断 dict 中 key

'''



class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if (s is None) or (len(s) == 0):
            return 0
        result , index = 1 , 0
        map = {}
        for i in range(len(s)):
            if s[i] in map:
                index = max(index, map[s[i]] + 1)
            map[s[i]] = i
            result = max(result, i - index + 1)
        return result

File: Solution/【003】无重复字符的最长子串/无重复字符的最长子串.cpp



class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if(s.empty() || s.size() == 0){
            return 0;
        }
        int index = 0;
        int result = 0;
        map<char, int> hashmap;
        for(int i = 0 ; i < s.length() ; i ++){
            char ch = s.at(i);
            if(hashmap.find(ch) != hashmap.end()){
                index = max(index, hashmap[ch] + 1);
            }
            hashmap[ch] = i;
            result = max(result, i - index + 1);
        }
        return result;
    }
};

